node/src/accountant/payment_adjuster/mod.rs

```diff
@@ -0,0 +1,2340 @@
+        let response_skeleton_opt = setup.response_skeleton_opt;
+        let agent = setup.agent;
+        let initial_service_fee_balance_minor = agent.service_fee_balance_minor();
+        let required_adjustment = setup.adjustment_analysis.adjustment;
+        let largest_exceeding_balance_recently_qualified =
+            find_largest_exceeding_balance(&analyzed_payables);
+
+        self.initialize_inner(
+            initial_service_fee_balance_minor,
+            required_adjustment,
+            largest_exceeding_balance_recently_qualified,
+            now,
+        );
+
+        let sketched_debug_info_opt = self.sketch_debug_info_opt(&analyzed_payables);
+
+        let affordable_accounts = self.run_adjustment(analyzed_payables)?;
+
+        self.complete_debug_info_if_enabled(sketched_debug_info_opt, &affordable_accounts);
+
+        self.reset_inner();
+
+        Ok(OutboundPaymentsInstructions::new(
+            Either::Right(affordable_accounts),
+            agent,
+            response_skeleton_opt,
+        ))
+    }
+
+    as_any_ref_in_trait_impl!();
+}
+
+impl Default for PaymentAdjusterReal {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl PaymentAdjusterReal {
+    pub fn new() -> Self {
+        Self {
+            analyzer: PreparatoryAnalyzer::new(),
+            disqualification_arbiter: DisqualificationArbiter::default(),
+            service_fee_adjuster: Box::new(ServiceFeeAdjusterReal::default()),
+            calculators: vec![Box::new(BalanceCriterionCalculator::default())],
+            inner: Box::new(PaymentAdjusterInnerNull::default()),
+            logger: Logger::new("PaymentAdjuster"),
+        }
+    }
+
+    fn initialize_inner(
+        &mut self,
+        cw_service_fee_balance: u128,
+        required_adjustment: Adjustment,
+        largest_exceeding_balance_recently_qualified: u128,
+        now: SystemTime,
+    ) {
+        let transaction_fee_limitation_opt = match required_adjustment {
+            Adjustment::TransactionFeeInPriority {
+                affordable_transaction_count,
+            } => Some(affordable_transaction_count),
+            Adjustment::ByServiceFee => None,
+        };
+
+        let inner = PaymentAdjusterInnerReal::new(
+            now,
+            transaction_fee_limitation_opt,
+            cw_service_fee_balance,
+            largest_exceeding_balance_recently_qualified,
+        );
+
+        self.inner = Box::new(inner);
+    }
+
+    fn reset_inner(&mut self) {
+        self.inner = Box::new(PaymentAdjusterInnerNull::default())
+    }
+
+    fn run_adjustment(
+        &mut self,
+        analyzed_accounts: Vec<AnalyzedPayableAccount>,
+    ) -> Result<Vec<PayableAccount>, PaymentAdjusterError> {
+        let weighted_accounts = self.calculate_weights(analyzed_accounts);
+        let processed_accounts = self.propose_adjustments_recursively(
+            weighted_accounts,
+            TransactionAndServiceFeeAdjustmentRunner {},
```

Comment:
Please take a look into the code to identify if we can keep either the enum `Adjustment` or this struct. They both are differentiating between the same thing but resulting in different combinations.

A proposed idea was to not use this `TransactionAndServiceFeeAdjustmentRunner {}` at all and instead use a boolean (checking if it's an initial call) with the old enum. You might need to use an `Either`.